jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py

```diff
@@ -85,14 +88,20 @@ def get_deployment_yamls(
         'shard_id': f'\"{shard_id}\"' if shard_id is not None else '\"\"',
         'pod_type': pod_type,
         'protocol': str(protocol).lower() if protocol is not None else '',
+        'volume_path': volumes[0] if volumes is not None else None
     }
 
     if gpus:
-        deployment_params['device_plugins'] = {'nvidia.com/gpu': gpus}
+        template_params['device_plugins'] = {'nvidia.com/gpu': gpus}
 
     template_name = 'deployment-executor' if name != 'gateway' else 'deployment-gateway'
 
-    if image_name_uses_before and image_name_uses_after:
+    if volumes is not None:
```

Comment:
I think volumes can be an empty list and in this case we don't want to do any of this
```suggestion
    if volumes:
```